crates/cargo-contract/src/cmd/config.rs

```diff
@@ -267,14 +266,41 @@ macro_rules! call_with_config {
             return res_nonspaced
         }
 
-        $crate::call_with_config_internal!(
+        let res_spaced = $crate::call_with_config_internal!(
             $obj,
             $function,
             format!("$crate :: cmd :: config :: {}", $config_name).as_str(),
             // All available chain configs need to be specified here
             $crate::cmd::config::Polkadot,
             $crate::cmd::config::Substrate,
             $crate::cmd::config::Ecdsachain
-        )
+        );
+        if !res_spaced.is_err() {
+            return res_spaced
+        }
+
+        let res_spaced_without_dollar = $crate::call_with_config_internal!(
+            $obj,
+            $function,
+            format!("crate :: cmd :: config :: {}", $config_name).as_str(),
+            // All available chain configs need to be specified here
+            $crate::cmd::config::Polkadot,
+            $crate::cmd::config::Substrate,
+            $crate::cmd::config::Ecdsachain
```

Comment:
To make this code a bit less brittle and easier to understand we could provide a mapping of the string from the config to the type e.g.

Then we are not dependent on the `stringify!` impl and we can remove the duplicate calls.

```suggestion
            "Polkadot" => $crate::cmd::config::Polkadot,
            "Substrate" => $crate::cmd::config::Substrate,
            "Ecdssachain" => $crate::cmd::config::Ecdsachain
```

Reply:
Thanks, Andrew! Great suggestion.